src/Module/BaseApi.php

```diff
@@ -195,80 +187,108 @@ public static function getRequest(array $defaults)
 	 *               'authenticated' => return status,
 	 *               'user_record' => return authenticated user record
 	 */
-	protected static function login(string $scope)
+	protected static function login(string $scope): bool
 	{
-		if (empty(self::$current_user_id)) {
-			self::$current_token = self::getTokenByBearer();
-			if (!empty(self::$current_token['uid'])) {
-				self::$current_user_id = self::$current_token['uid'];
-			} else {
-				self::$current_user_id = 0;
-			}
+		if (!self::getCachedCurrentUserIDFromRequest()) {
+			Logger::debug(API_LOG_PREFIX . 'failed', ['module' => 'api', 'action' => 'login', 'parameters' => $_SERVER]);
+			header('WWW-Authenticate: Basic realm="Friendica"');
+			throw new HTTPException\UnauthorizedException('This API requires login');
 		}
 
-		if (!empty($scope) && !empty(self::$current_token)) {
-			if (empty(self::$current_token[$scope])) {
-				Logger::warning('The requested scope is not allowed', ['scope' => $scope, 'application' => self::$current_token]);
-				DI::mstdnError()->Forbidden();
-			}
-		}
-
-		if (empty(self::$current_user_id)) {
-			// The execution stops here if no one is logged in
-			api_login(DI::app());
+		$current_token = self::getCachedTokenByBearer();
+		if ($current_token && empty($current_token[$scope])) {
```

Comment:
Why have you removed the check for an empty scope? When we don't provide a scope for the login we shouldn't check it, I think.

Reply:
None of the current calls to this method have an empty scope and the parameter didn't have a default empty value either, so I figured it was some sort of implicit constraint that came later during development but the condition was never removed. In the current codebase the check for empty scope condition will never be hit.

Reply:
When we say that we always need to provide a scope, we should stop with `DI::mstdnError()->Forbidden();` on an empty value as a precaution, shouldn't we? Otherwise we would just create some PHP warning or notice on an invalid scope.

Reply:
Yes, PHP will barf if someone tries to call `checkAllowedScope(null)`, but since we don't pass user-supplied data in this parameter, we don't have to account for it. Only developers can provide a wrong value, and it will instantly prevent their code from working.